stream-chat-android/src/main/java/com/getstream/sdk/chat/ChatUI.kt

```diff
@@ -29,66 +25,57 @@ import io.getstream.chat.android.livedata.ChatDomain
  * @see ChatFonts
  */
 public class ChatUI internal constructor(
-    public val client: ChatClient,
-    public val chatDomain: ChatDomain,
     public val fonts: ChatFonts,
     public val strings: ChatStrings,
-    public val navigationHandler: ChatNavigationHandler? = null,
+    public val navigator: ChatNavigator,
     public val markdown: ChatMarkdown,
     public val urlSigner: UrlSigner
 ) {
     public val version: String
         get() = BuildConfig.BUILD_TYPE + ":" + BuildConfig.STREAM_CHAT_UI_VERSION
 
-    public val navigator: ChatNavigator = ChatNavigatorImpl()
+    public class Builder(private val appContext: Context) {
 
-    init {
-        if (navigationHandler != null) {
-            navigator.setHandler(navigationHandler)
-        }
-    }
-
-    public data class Builder(
-        private var client: ChatClient,
-        private var chatDomain: ChatDomain,
-        private var appContext: Context
-    ) {
```

Comment:
Changing this constructor signature is a significant breaking change for anyone who's initializing our library with any of the recent versions. I think it would be a very good idea to keep around a 3-param deprecated constructor and provide a migration path to this new API.

Or, at the very least, add more details to the release notes. But I'm really not a fan of breaking the library init for _all_ clients.

Reply:
Because it is a builder, I can keep the old constructor, doing nothing with `ChatClient` and `ChatDomain` params ;)

Reply:
Yup, that's what I was thinking. Plus we can add a nice `@Deprecated` annotation and `ReplaceWith("Builder(appContext)")`

Reply:
That was what I was thinking about ;)
DONE